python_modules/libraries/dagster-dg/dagster_dg/cli/component.py

```diff
@@ -259,10 +272,76 @@ def component_check_command(
     **global_options: object,
 ) -> None:
     """Check component files against their schemas, showing validation errors."""
+    resolved_paths = [Path(path).absolute() for path in paths]
+    top_level_component_validator = Draft202012Validator(schema=COMPONENT_FILE_SCHEMA)
+
     cli_config = normalize_cli_config(global_options, context)
     dg_context = DgContext.from_config_file_discovery_and_cli_config(Path.cwd(), cli_config)
 
-    try:
-        dg_context.external_components_command(["check", "component", *paths])
-    except CalledProcessError:
-        sys.exit(1)
+    component_registry = RemoteComponentRegistry.from_dg_context(dg_context)
+
+    validation_errors: list[tuple[Optional[str], ValidationError]] = []
+
+    component_contents_by_dir = {}
+    local_components = set()
+    for component_dir in (
+        dg_context.root_path / dg_context.root_package_name / "components"
```

Comment:
you can simplify this a bit with:

```python
for component_path in component_path.rglob("component.yaml"):
    ...
```

Reply:
that'll handle walking the directory for you and finding all the component.yaml files. You can probably also use rglob to filter for things in the resolved paths, although not sure if that's that much better

Reply:
I bias a little towards the iterdir approach just to mirror 1:1 how we actually pick up components elsewhere in the codebase - we end up needing to find both the component yaml and the component dir anyway so it doesn't save a ton of space

Reply:
seems like a case where a common utility would be nice here -- nbd for now